src/pull-request-core.ts

```diff
@@ -0,0 +1,234 @@
+import Octokit, { GitUpdateRefResponse } from '@octokit/rest';
+
+import {
+    PullRequest,
+    PullRequestWithChangesetCommit,
+    SourceFile
+} from './types';
+
+export function issuePullRequestWithToken(
+    { pullRequest, changesetCommit }: PullRequestWithChangesetCommit,
+    githubAccessToken: string
+): Promise<Octokit.PullsCreateResponse> {
+    const octokit = new Octokit({
+        auth: githubAccessToken
+    });
+
+    return Promise.resolve(issuePullRequest({ pullRequest, changesetCommit }));
+
+    function issuePullRequest({
+        pullRequest: pr,
+        changesetCommit: cs
+    }: PullRequestWithChangesetCommit): Promise<Octokit.PullsCreateResponse> {
+        return Promise.resolve(
+            getCommitBranchSha(
+                pr.sourceOwner,
+                pr.sourceRepo,
+                pr.destinationBranch,
+                pr.sourceBranch
+            )
+                .then(commitBranchSha =>
+                    createTree(
+                        pr.sourceOwner,
+                        pr.sourceRepo,
+                        commitBranchSha,
+                        cs.sourceFiles
+                    )
+                )
+                .then(([treeSha, commitBranchSha]) =>
+                    pushCommit(
+                        pr.sourceOwner,
+                        pr.sourceRepo,
+                        pr.sourceBranch,
+                        commitBranchSha,
+                        treeSha,
+                        cs.authorName,
+                        cs.authorEmail,
+                        cs.commitMessage
+                    )
+                )
+                .then(_ => createPullRequest(pr))
```

Comment:
This will create no parameter binding, instead of a parameter named `_`
```suggestion
                .then(() => createPullRequest(pr))
```